### ISSUE TITLE
Segfault when using :undo with view-source tab

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#2289]
This crashes: `python3 -m qutebrowser --backend webengine  --temp-basedir --debug 'qutebrowser.org' 'view-source:qutebrowser.org'  ':later 1000 tab-close' ':later 2000 undo'`

Output:

```
22:52:51 DEBUG    commands   command:run:524 Calling qutebrowser.browser.commands.CommandDispatcher.undo(<qutebrowser.browser.commands.CommandDispatcher>)
22:52:51 DEBUG    webview    tabbedbrowser:tabopen:382 Creating new tab with URL PyQt5.QtCore.QUrl('view-source:http://qutebrowser.org/'), background False, explicit False, idx 1
[...]
22:52:51 DEBUG    mouse      mouse:eventFilter:56 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7f540c0c4288> got new child <PyQt5.QtWidgets.QWidget object at 0x7f540c0e91f8>, installing filter
[...]
22:52:51 DEBUG    mouse      mouse:eventFilter:56 <qutebrowser.browser.webengine.webview.WebEngineView object at 0x7f540c0c4288> got new child <PyQt5.QtWidgets.QWidget object at 0x7f540c0e93a8>, installing filter
[31997:31997:0100/000000:ERROR:broker_posix.cc(41)] Invalid node channel message
[32001:32001:0100/000000:ERROR:broker_posix.cc(41)] Invalid node channel message
```

Stack:

```
#0  0x00007fffd8bb7e06 in QtWebEngineCore::WebContentsViewQt::RenderViewCreated(content::RenderViewHost*) (this=0x3f85d30, host=<optimized out>)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/core/web_contents_view_qt.cpp:99
#1  0x00007fffd9407186 in content::WebContentsImpl::RenderViewCreated(content::RenderViewHost*) (this=0x3ff77e0, render_view_host=0x48048b0)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/web_contents/web_contents_impl.cc:4193
#2  0x00007fffd936f705 in content::RenderViewHostImpl::CreateRenderView(int, int, int, content::FrameReplicationState const&, bool) (this=0x48048b0, opener_frame_route_id=<optimized out>, proxy_route_id=<optimized out>, max_page_id=<optimized out>, replicated_frame_state=..., window_was_created_with_opener=<optimized out>)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/renderer_host/render_view_host_impl.cc:394
#3  0x00007fffd941256a in content::WebContentsImpl::CreateRenderViewForRenderManager(content::RenderViewHost*, int, int, content::FrameReplicationState const&) (this=0x3ff77e0, render_view_host=0x48048b0, opener_frame_routing_id=<optimized out>, proxy_routing_id=-2, replicated_frame_state=...) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/web_contents/web_contents_impl.cc:4824
#4  0x00007fffd92ef190 in content::RenderFrameHostManager::CreateRenderFrame(content::SiteInstance*, bool, int*) (this=this@entry=0x3f7bb80, instance=instance@entry=0x3f842d0, hidden=<optimized out>, view_routing_id_ptr=view_routing_id_ptr@entry=0x0) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/render_frame_host_manager.cc:1714
#5  0x00007fffd92f18dc in content::RenderFrameHostManager::CreatePendingRenderFrameHost(content::SiteInstance*, content::SiteInstance*) (this=0x3f7bb80, old_instance=0x3f7bf60, new_instance=0x3f842d0)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/render_frame_host_manager.cc:1563
#6  0x00007fffd92f3790 in content::RenderFrameHostManager::UpdateStateForNavigate(GURL const&, content::SiteInstance*, content::SiteInstance*, ui::PageTransition, bool, bool, content::GlobalRequestID const&, int, bool) (this=0x3f7bb80, dest_url=..., source_instance=<optimized out>, dest_instance=<optimized out>, transition=<optimized out>, dest_is_restore=<optimized out>, dest_is_view_source_mode=<optimized out>, transferred_request_id=..., bindings=<optimized out>, is_reload=<optimized out>) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/render_frame_host_manager.cc:2208
#7  0x00007fffd92f3996 in content::RenderFrameHostManager::Navigate(GURL const&, content::FrameNavigationEntry const&, content::NavigationEntryImpl const&, bool) (this=this@entry=0x3f7bb80, dest_url=..., frame_entry=..., entry=
    ..., is_reload=<optimized out>) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/render_frame_host_manager.cc:212
#8  0x00007fffd95164d6 in content::NavigatorImpl::NavigateToEntry(content::FrameTreeNode*, content::FrameNavigationEntry const&, content::NavigationEntryImpl const&, content::NavigationController::ReloadType, bool, bool, scoped_refptr<content::ResourceRequestBodyImpl> const&) (this=this@entry=0x15091a0, frame_tree_node=frame_tree_node@entry=0x3f7bb70, frame_entry=..., entry=..., reload_type=reload_type@entry=content::NavigationController::NO_RELOAD, is_same_document_history_load=is_same_document_history_load@entry=false, is_pending_entry=<optimized out>, post_body=...)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/navigator_impl.cc:366
#9  0x00007fffd9516b4b in content::NavigatorImpl::NavigateToPendingEntry(content::FrameTreeNode*, content::FrameNavigationEntry const&, content::NavigationController::ReloadType, bool) (this=0x15091a0, frame_tree_node=0x3f7bb70, frame_entry=..., reload_type=content::NavigationController::NO_RELOAD, is_same_document_history_load=<optimized out>)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/navigator_impl.cc:453
#10 0x00007fffd92d30bb in content::NavigationControllerImpl::NavigateToPendingEntryInternal(content::NavigationController::ReloadType) (this=this@entry=0x3ff7870, reload_type=reload_type@entry=content::NavigationController::NO_RELOAD)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/navigation_controller_impl.cc:1822
#11 0x00007fffd92d3335 in content::NavigationControllerImpl::NavigateToPendingEntry(content::NavigationController::ReloadType) (this=0x3ff7870, reload_type=content::NavigationController::NO_RELOAD)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/3rdparty/chromium/content/browser/frame_host/navigation_controller_impl.cc:1798
#12 0x00007fffd8bad7fc in QtWebEngineCore::WebContentsAdapter::initialize(QtWebEngineCore::WebContentsAdapterClient*) (this=0x3f83e20, adapterClient=adapterClient@entry=0x5dd5650)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/core/web_contents_adapter.cpp:424
#13 0x00007fffddf2b0ae in QWebEnginePagePrivate::recreateFromSerializedHistory(QDataStream&) (this=0x5dd5650, input=...)
    at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/webenginewidgets/api/qwebenginepage.cpp:639
#14 0x00007fffddf27fc2 in operator>>(QDataStream&, QWebEngineHistory&) (stream=..., history=...) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-opensource-src-5.8.0/src/webenginewidgets/api/qwebenginehistory.cpp:273
#15 0x00007fffde172050 in slot___rshift__(PyObject*, PyObject*) (sipArg0=0x7fffb810ae48, sipArg1=0x7fffb810a518) at sipQtWebEngineWidgetscmodule.cpp:971
#16 0x00007ffff0dcb2df in  () at /usr/lib/python3.6/site-packages/sip.so
#17 0x00007ffff5c98a6a in slot_QDataStream___rshift__(PyObject*, PyObject*) (sipArg0=0x7fffb810ae48, sipArg1=0x7fffb810a518) at sipQtCoreQDataStream.cpp:2828
#18 0x00007ffff7398943 in  () at /usr/lib/libpython3.6m.so.1.0
#19 0x00007ffff747b616 in PyNumber_Rshift () at /usr/lib/libpython3.6m.so.1.0
#20 0x00007ffff73e5a22 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#21 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#22 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#23 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#24 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#25 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#26 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#27 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#28 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#29 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#30 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#31 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#32 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#33 0x00007ffff7427f47 in PyEval_EvalCodeEx () at /usr/lib/libpython3.6m.so.1.0
#34 0x00007ffff74428d5 in  () at /usr/lib/libpython3.6m.so.1.0
#35 0x00007ffff747d91b in PyObject_Call () at /usr/lib/libpython3.6m.so.1.0
#36 0x00007ffff73e557e in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#37 0x00007ffff7426291 in  () at /usr/lib/libpython3.6m.so.1.0
#38 0x00007ffff742778a in  () at /usr/lib/libpython3.6m.so.1.0
#39 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#40 0x00007ffff73e4ba9 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#41 0x00007ffff7426291 in  () at /usr/lib/libpython3.6m.so.1.0
#42 0x00007ffff742778a in  () at /usr/lib/libpython3.6m.so.1.0
#43 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#44 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#45 0x00007ffff7426291 in  () at /usr/lib/libpython3.6m.so.1.0
#46 0x00007ffff7427074 in _PyFunction_FastCallDict () at /usr/lib/libpython3.6m.so.1.0
#47 0x00007ffff747ca4e in _PyObject_FastCallDict () at /usr/lib/libpython3.6m.so.1.0
#48 0x00007ffff747d852 in _PyObject_Call_Prepend () at /usr/lib/libpython3.6m.so.1.0
#49 0x00007ffff747c84c in _PyObject_FastCallDict () at /usr/lib/libpython3.6m.so.1.0
#50 0x00007ffff749b13a in  () at /usr/lib/libpython3.6m.so.1.0
#51 0x00007ffff747d91b in PyObject_Call () at /usr/lib/libpython3.6m.so.1.0
#52 0x00007ffff5db2f56 in PyQtSlot::call(_object*, _object*) const (this=0x2d0c520, callable=0x7fffb8058408, args=0x7ffff7f82048) at ../qpy/QtCore/qpycore_pyqtslot.cpp:247
#53 0x00007ffff5db2c51 in PyQtSlot::invoke(void**, _object*, void*, bool) const (this=0x2d0c520, qargs=0x7fffffffc7c0, self=0x0, result=0x0, no_receiver_check=false) at ../qpy/QtCore/qpycore_pyqtslot.cpp:159
#54 0x00007ffff5db2956 in PyQtSlot::invoke(void**, bool) const (this=0x2d0c520, qargs=0x7fffffffc7c0, no_receiver_check=false) at ../qpy/QtCore/qpycore_pyqtslot.cpp:78
#55 0x00007ffff5db16b8 in PyQtSlotProxy::unislot(void**) (this=0x4032040, qargs=0x7fffffffc7c0) at ../qpy/QtCore/qpycore_pyqtslotproxy.cpp:205
#56 0x00007ffff5db160d in PyQtSlotProxy::qt_metacall(QMetaObject::Call, int, void**) (this=0x4032040, _c=QMetaObject::InvokeMetaMethod, _id=0, _a=0x7fffffffc7c0) at ../qpy/QtCore/qpycore_pyqtslotproxy.cpp:170
#57 0x00007ffff576c354 in QMetaObject::activate(QObject*, int, int, void**) (sender=sender@entry=0x150c3d0, signalOffset=<optimized out>, local_signal_index=local_signal_index@entry=0, argv=argv@entry=0x0) at kernel/qobject.cpp:3755
#58 0x00007ffff576cad7 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) (sender=sender@entry=0x150c3d0, m=m@entry=0x7ffff5b8e780 <QTimer::staticMetaObject>, local_signal_index=local_signal_index@entry=0, argv=argv@entry=0x0) at kernel/qobject.cpp:3602
#59 0x00007ffff57e6750 in QTimer::timeout(QTimer::QPrivateSignal) (this=this@entry=0x150c3d0) at .moc/moc_qtimer.cpp:200
#60 0x00007ffff5779208 in QTimer::timerEvent(QTimerEvent*) (this=0x150c3d0, e=<optimized out>) at kernel/qtimer.cpp:254
#61 0x00007ffff5d5b728 in sipQTimer::timerEvent(QTimerEvent*) (this=0x150c3d0, a0=0x7fffffffcb90) at sipQtCoreQTimer.cpp:187
#62 0x00007ffff576cfd3 in QObject::event(QEvent*) (this=0x150c3d0, e=<optimized out>) at kernel/qobject.cpp:1285
#63 0x00007ffff5d5b5f8 in sipQTimer::event(QEvent*) (this=0x150c3d0, a0=0x7fffffffcb90) at sipQtCoreQTimer.cpp:156
#64 0x00007fffe01883dc in QApplicationPrivate::notify_helper(QObject*, QEvent*) (this=<optimized out>, receiver=0x150c3d0, e=0x7fffffffcb90) at kernel/qapplication.cpp:3745
#65 0x00007fffe018fbf1 in QApplication::notify(QObject*, QEvent*) (this=0xdd6f40, receiver=0x150c3d0, e=0x7fffffffcb90) at kernel/qapplication.cpp:3502
#66 0x00007fffe0a06bfe in sipQApplication::notify(QObject*, QEvent*) (this=0xdd6f40, a0=0x150c3d0, a1=0x7fffffffcb90) at sipQtWidgetsQApplication.cpp:232
#67 0x00007ffff57408b0 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (receiver=0x150c3d0, event=event@entry=0x7fffffffcb90) at kernel/qcoreapplication.cpp:988
#68 0x00007ffff579412e in QCoreApplication::sendEvent(QObject*, QEvent*) (event=0x7fffffffcb90, receiver=<optimized out>) at ../../include/QtCore/../../src/corelib/kernel/qcoreapplication.h:231
#69 0x00007ffff579412e in QTimerInfoList::activateTimers() (this=0xf1f780) at kernel/qtimerinfo_unix.cpp:643
#70 0x00007ffff57949b9 in timerSourceDispatch (source=<optimized out>) at kernel/qeventdispatcher_glib.cpp:182
#71 0x00007ffff57949b9 in idleTimerSourceDispatch(GSource*, GSourceFunc, gpointer) (source=<optimized out>) at kernel/qeventdispatcher_glib.cpp:229
#72 0x00007ffff2015587 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#73 0x00007ffff20157f0 in  () at /usr/lib/libglib-2.0.so.0
#74 0x00007ffff201589c in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#75 0x00007ffff57954bf in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (this=0xf1f520, flags=...) at kernel/qeventdispatcher_glib.cpp:423
#76 0x00007ffff573ed0a in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (this=this@entry=0x7fffffffcde0, flags=..., flags@entry=...) at kernel/qeventloop.cpp:212
#77 0x00007ffff5747254 in QCoreApplication::exec() () at kernel/qcoreapplication.cpp:1261
#78 0x00007fffe0a094a1 in meth_QApplication_exec_(PyObject*, PyObject*) (sipArgs=0x7ffff7f82048) at sipQtWidgetsQApplication.cpp:1506
#79 0x00007ffff741891f in _PyCFunction_FastCallDict () at /usr/lib/libpython3.6m.so.1.0
#80 0x00007ffff7427a0f in  () at /usr/lib/libpython3.6m.so.1.0
#81 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#82 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#83 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#84 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#85 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#86 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#87 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#88 0x00007ffff742753a in  () at /usr/lib/libpython3.6m.so.1.0
#89 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#90 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#91 0x00007ffff7427f47 in PyEval_EvalCodeEx () at /usr/lib/libpython3.6m.so.1.0
#92 0x00007ffff73e3a5b in PyEval_EvalCode () at /usr/lib/libpython3.6m.so.1.0
#93 0x00007ffff73db871 in  () at /usr/lib/libpython3.6m.so.1.0
#94 0x00007ffff741891f in _PyCFunction_FastCallDict () at /usr/lib/libpython3.6m.so.1.0
#95 0x00007ffff7427a0f in  () at /usr/lib/libpython3.6m.so.1.0
#96 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#97 0x00007ffff7426291 in  () at /usr/lib/libpython3.6m.so.1.0
#98 0x00007ffff742778a in  () at /usr/lib/libpython3.6m.so.1.0
#99 0x00007ffff7427af3 in  () at /usr/lib/libpython3.6m.so.1.0
#100 0x00007ffff73e3d77 in _PyEval_EvalFrameDefault () at /usr/lib/libpython3.6m.so.1.0
#101 0x00007ffff7427f47 in PyEval_EvalCodeEx () at /usr/lib/libpython3.6m.so.1.0
#102 0x00007ffff7442838 in  () at /usr/lib/libpython3.6m.so.1.0
#103 0x00007ffff747d91b in PyObject_Call () at /usr/lib/libpython3.6m.so.1.0
#104 0x00007ffff74a3cbd in  () at /usr/lib/libpython3.6m.so.1.0
#105 0x00007ffff74a43aa in Py_Main () at /usr/lib/libpython3.6m.so.1.0
#106 0x0000000000400a5d in main ()
```

I suspect https://codereview.qt-project.org/#/c/181111/ will help, but maybe there's something we can do.

**Comment by The-Compiler** [comment]
This regressed somehow.

**Comment by The-Compiler** [comment]
This regressed with 98f17a03bbca0ff7369eeb4a1c677caf7090d42f which fixes another crash. However, it's fixed with Qt 5.9.
